config/initializers/secure_headers.rb

```diff
@@ -72,36 +71,8 @@
   # }
 end
 
-# A tiny middleware that calls a block on each request. When both:
-# 1) the block returns true
-# 2) the response is a 2XX response
-# It deletes the Content-Security-Policy header. This is intended so that we can override
-# SecureHeaders behavior and not set the headers on asset files, because the headers should be set
-# on the document that links to the assets, not the assets themselves.
-class SecureHeaders::RemoveContentSecurityPolicy
-  # @yieldparam [Rack::Request] request
-  def initialize(app, &block)
-    @app = app
-    @block = block
-  end
-
-  def call(env)
-    status, headers, body = @app.call(env)
-
-    if (200...300).cover?(status) && @block.call(Rack::Request.new(env))
-      headers.delete('Content-Security-Policy')
-    end
-
-    [status, headers, body]
-  end
-end
-
-# We need this to be called after the SecureHeaders::Railtie adds its own middleware at the top
-Rails.application.configure do |config|
-  config.middleware.insert_before(
-    SecureHeaders::Middleware,
-    SecureHeaders::RemoveContentSecurityPolicy,
-  ) do |request|
-    request.path.start_with?('/acuant/')
+if FeatureManagement.rails_csp_tooling_enabled?
+  Rails.application.configure do |config|
+    config.middleware.delete SecureHeaders::Middleware
```

Comment:
Will we still want to use `SecureHeaders` for its non-CSP-related features?

Reply:
I'm still debating this. The issue with keeping the middleware is if we don't keep the configuration it sets some defaults here. I'm gonna look to see if there is an easy way we can make it not do that. If not then we may need to go all or nothing with it.

Thankfully, the other things it does, setting headers and adding things to cookies, are pretty straightforward to do without it.

Reply:
Looks like we may be able to set an opt out option for the CSP so I think we may be able to keep it for the rest of the stuff.